service/matching/forwarder.go

```diff
@@ -154,6 +154,7 @@ func (fwdr *Forwarder) ForwardTask(ctx context.Context, task *internalTask) erro
 				ScheduleToStartTimeout: expirationDuration,
 				ForwardInfo:            fwdr.getForwardInfo(task),
 				VersionDirective:       task.event.Data.GetVersionDirective(),
+				Priority:               task.event.Data.Priority,
```

Comment:
I'd use GetPriority() here just to be sure. I forgot if there were any circumstances where `task.event.Data` can be nil but the above lines do it